tests/components/iotty/test_api.py

```diff
@@ -0,0 +1,79 @@
+"""Unit tests for iottycloud API."""
+
+from unittest.mock import patch
+
+from aiohttp import ClientSession
+import pytest
+
+from homeassistant.components.iotty import api
+from homeassistant.components.iotty.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from tests.common import MockConfigEntry
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def test_api_create_fail(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test API creation with no session."""
+
+    with pytest.raises(ValueError) as excinfo:
+        _ = api.IottyProxy(hass, None, None)
+    assert "websession" in str(excinfo.value)
```

Comment:
Note that you can pass a regex which is matched against the stringified exception: https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.raises
Also, there shouldn't be a need to assign to `_` here?

Reply:
I didn't know of that option; implemented like that